src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,415 @@
+            elif value[_TYPE] == _SETTLED:
+                # ignorable in terms of in/out/intra
+                pass
+            elif value[_TYPE] == _CREDIT:
+                # 'credit not implemented'
+                pass
+            elif value[_TYPE] == _STAKING:
+                # 'staking not implemented'
+                pass
+            elif value[_TYPE] == _SALE:
+                # 'sale not implemented'
+                pass
+            else:
+                unhandled_types.update({value[_TYPE]: key})
+        self.__logger.debug(f"unhandled types of the ledger={unhandled_types}")
+
+        return self._compute_tx_set(trade_history, ledger, result)
+
+    def _compute_tx_set(self, trade_history, ledger, processed_transactions) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        transactions: List[AbstractTransaction] = processed_transactions[_IN] + \
+                                                  processed_transactions[_OUT] + \
+                                                  processed_transactions[_INTRA]
+        for key in transactions:
+            record: Dict[str, str] = ledger[key]
+            self.__logger.debug("Ledger record: %s", record)
+
+            timestamp_value: str = self._rp2_timestamp_from_ms_epoch(str(RP2Decimal(record[_TIMESTAMP])*RP2Decimal(_MS_IN_SECOND)))
+
+            is_fiat_asset: bool = record[_ASSET] in _FIAT_SET or 'USD' in record[_ASSET]
+
+            amount: RP2Decimal = abs(RP2Decimal(record[_AMOUNT]))
+            asset: str = self.baseId_to_base[record[_ASSET]]
+            kwargs: Dict[str, str] = {
+                'unique_id': Keyword.UNKNOWN.value,
+                'plugin': self.__PLUGIN_NAME,
+                'raw_data': str(record),
+                'timestamp': str(timestamp_value),
+                'spot_price': '0',
+                'asset': asset,
+                'notes': key,
+            }
+
+            if record[_TYPE] == _WITHDRAWAL or record[_TYPE] == _DEPOSIT:
+                is_deposit: bool = record[_TYPE] == _DEPOSIT
+                is_withdrawal: bool = record[_TYPE] == _WITHDRAWAL
+
+                # Intra
+                kwargs.update(
+                    {
+                        'unique_id': Keyword.UNKNOWN.value,
+                        'from_exchange': self.__EXCHANGE_NAME if is_withdrawal else Keyword.UNKNOWN.value,
+                        'from_holder': self.account_holder if is_withdrawal else Keyword.UNKNOWN.value,
+                        'to_exchange': self.__EXCHANGE_NAME if is_deposit else Keyword.UNKNOWN.value,
+                        'to_holder': self.account_holder if is_deposit else Keyword.UNKNOWN.value,
+                        'crypto_sent': Keyword.UNKNOWN.value if is_deposit else str(amount),
+                        'crypto_received': str(amount) if is_deposit else Keyword.UNKNOWN.value,
+                    }
+                )
+                result.append(IntraTransaction(**kwargs))
```

Comment:
I recommend not using an argument dictionary: see my other comment below for more details.

Reply:
What would be the preferred way? I'd like to keep it but I also would like to not deviate if there is a style choice in mind.

Reply:
The preferred way is to just spell out the parameters in the instantiation without using kwargs. Here's an [example from the Coinbase plugin](https://github.com/eprbell/dali-rp2/blob/main/src/dali/plugin/input/rest/coinbase.py#L343). The reason is that reusing the dictionary can lead to subtle and hard-to-fix bugs: it's not worth the line number savings in the code in my opinion.